src/app/banking/vouchers/view-voucher.component.ts

```diff
@@ -15,39 +20,136 @@ import { Menu } from 'app/shared/menu';
 export class ViewVoucherComponent extends BaseViewPageComponent<VoucherView> implements OnInit {
 
   qrCodeUrl$ = new BehaviorSubject<string>(null);
+  createCancelDeviceConfirmation: () => CreateDeviceConfirmation;
+  createChangeExpirationDeviceConfirmation: () => CreateDeviceConfirmation;
+  canConfirm: boolean;
+  confirmationPassword: FormControl;
+  viewExpirationDate: boolean;
 
   constructor(
     injector: Injector,
     private voucherService: VouchersService,
     public bankingHelper: BankingHelperService
-  ) {
-    super(injector);
-  }
+  ) { super(injector); }
 
   ngOnInit() {
     super.ngOnInit();
     const key = this.route.snapshot.paramMap.get('key');
-    this.addSub(this.voucherService.viewVoucher({ key: key }).subscribe(voucher => {
-      this.data = voucher;
-      this.addSub(this.voucherService.getVoucherQrCode({ key: voucher.id, size: ImageSizeEnum.MEDIUM })
-        .subscribe(image => this.qrCodeUrl$.next(URL.createObjectURL(image))));
-      this.headingActions = this.initActions();
-    }));
+    this.addSub(this.voucherService.viewVoucher({ key: key }).subscribe(voucher => this.data = voucher));
+  }
+
+  onDataInitialized(_data) {
+    this.addSub(this.voucherService.getVoucherQrCode({ key: _data.id, size: ImageSizeEnum.MEDIUM })
+      .subscribe(image => this.qrCodeUrl$.next(URL.createObjectURL(image))));
+    this.createCancelDeviceConfirmation = () => ({
+      type: DeviceConfirmationTypeEnum.MANAGE_VOUCHER,
+      voucher: _data.id,
+      voucherAction: VoucherActionEnum.CANCEL
+    });
+    this.createChangeExpirationDeviceConfirmation = () => ({
+      type: DeviceConfirmationTypeEnum.MANAGE_VOUCHER,
+      voucher: _data.id,
+      voucherAction: VoucherActionEnum.CHANGE_EXPIRATION
+    });
+    this.canConfirm = this.authHelper.canConfirm(_data.confirmationPasswordInput);
+    if (_data.confirmationPasswordInput) {
+      this.confirmationPassword = this.formBuilder.control('confirmationPassword', Validators.required);
+    }
+    this.headingActions = this.initActions(_data);
+    this.viewExpirationDate = _data.expirationDate && _data.status !== VoucherStatusEnum.CANCELED
```

Comment:
Once https://jira.cyclos.org/browse/CYCLOS-7717 is implemented, we no longer need to check for the status to show the expiration date, but test if it has a value.